java/src/main/java/com/genexus/filters/APIObjectFilter.java

```diff
@@ -60,6 +62,15 @@ public void init(FilterConfig filterConfig) throws ServletException {
             String paramValue = config.getInitParameter("BasePath");
             if (paramValue != null && !paramValue.isEmpty())
             {
+                logger.info("API basepath parameter: " +  paramValue) ;
+                if (paramValue == "*")
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: An equality operator (== or !=) is used to compare strings, which matches strings based on address.

**Fix**: Do content comparison using the equals() method to compare the values of the strings.

Reply:
Changed it. thanks.